openadapt/privacy/providers/private_ai.py

```diff
@@ -0,0 +1,191 @@
+            "link_batch": False,
+            "entity_detection": {
+                "accuracy": "high",
+                "return_entity": True,
+            },
+            "processed_text": {
+                "type": "MARKER",
+                "pattern": "[UNIQUE_NUMBERED_ENTITY_TYPE]",
+            },
+        }
+
+        headers = {
+            "Content-Type": HEADER_CONTENT_TYPE,
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(TEXT_URL, json=payload, headers=headers)
+
+        data = response.json()
+        logger.debug(f"{data=}")
+
+        # According to the API documentation,
+        # the response is a list of dicts when there is no error/issue in the request
+        # else it is a dict with a key "detail" containing the error message
+
+        if type(data) is dict and "detail" in data:
+            raise ValueError(data.get("detail"))
+
+        redacted_text = data[0].get("processed_text")
+        logger.debug(f"{redacted_text=}")
+
+        return redacted_text
+
+    def scrub_image(
+        self,
+        image: Image,
+        fill_color: int = config.SCRUB_FILL_COLOR,  # pylint: disable=no-member
+    ) -> Image:
+        """Scrub the image of all PII/PHI.
+
+        Args:
+            image (Image): A PIL.Image object to be redacted
+            fill_color (int): The color used to fill the redacted regions(BGR).
+
+        Returns:
+            Image: The redacted image with PII and PHI removed.
+        """
+        buffer = BytesIO()
+
+        image.save(buffer, format="PNG")
+        # Get the image data as bytes
+        image_data = buffer.getvalue()
+
+        file_data = base64.b64encode(image_data)
+        file_data = file_data.decode("ascii")
+
+        # Clean up by closing the BytesIO buffer
+        buffer.close()
+
+        payload = {
+            "file": {"data": file_data, "content_type": IMAGE_CONTENT_TYPE},
+            "entity_detection": {"accuracy": "high", "return_entity": True},
+            "pdf_options": {"density": 150, "max_resolution": 2000},
+            "audio_options": {"bleep_start_padding": 0, "bleep_end_padding": 0},
+        }
+
+        headers = {
+            "Content-Type": HEADER_CONTENT_TYPE,
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(BASE64_URL, json=payload, headers=headers)
+        response = response.json()
+        logger.debug(f"{response=}")
+
+        # According to the API documentation,
+        # the response is a list of dicts when there is no error/issue in the request
+        # else it is a dict with a key "detail" containing the error message
+
+        if type(response) is dict and "detail" in response:
+            raise ValueError(response.get("detail"))
+
+        redact_file_path = os.path.join(FILES_DIR, f"redacted-{TEMP_IMAGEFILE_NAME}")
+
+        # Write to file
+        with open(redact_file_path, "wb") as redacted_file:
```

Comment:
What do you think about renaming `redact_file_path` to `redacted_file_path` to be consistent with `redacted_file_data`?

Also, what do you think about using BytesIO here as well, and elsewhere a file is written in this PR?

Reply:
>> Also, what do you think about using BytesIO here as well, and elsewhere a file is written in this PR?

Addressed in: [2f92260](https://github.com/OpenAdaptAI/OpenAdapt/pull/486/commits/2f922602e139e32bead068caf961238b23fb3b6a)